projects/core/src/cms/services/cms-structure-config.service.ts

```diff
@@ -0,0 +1,135 @@
+import { Injectable } from '@angular/core';
+import {
+  CmsStructureConfig,
+  CmsPageConfig,
+  CmsPageSlotsConfig
+} from '../config/cms-structure.config';
+import { CmsStructureModel } from '../model/page.model';
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+/**
+ * Service that provides access to CMS structure from a static
+ * configuration or configuration file. This class uses static
+ * configuration is designed in async fashion so that configuratiosn
+ * can be loaded from a file or stream.
+ *
+ * The intend of the `CmsStructureConfigService` however is to provide
+ * fast loading pages and default cms structure for comodoty commerce.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export abstract class CmsStructureConfigService {
```

Comment:
Some tests for this class, especially for merge logic, would probably be beneficial.